### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by amit-negi-23**
homepage demo

### CODE REVIEW

File: src/components/home/home.jsx [hunk 1]
```diff
@@ -1,45 +1,83 @@
...
+        }
+        const data = await response.json();
+        console.log("aa", data)
+        setBlogPosts(data.posts.slice(0, 6));
+        setLoading(false);
+      } catch (error) {
+        setError(error.message);
+        setLoading(false);
+      }
+    };
```
Comment (Mansibansal9648) [inline src/components/home/home.jsx:36]: we can define fetchBlogPosts function separately

File: src/components/home/home.jsx [hunk 2]
```diff
@@ -1,45 +1,83 @@
...
-  }, []);
+  }, [])
 
-  function onChangeHandler(event) {
-    setUser({
-      ...user,
-      [event.target.name]: event.target.value
-    })
-  }
+  useEffect(() => {
```
Comment (Mansibansal9648) [inline src/components/home/home.jsx:21]: why we are using 2 different useeffects?
can't we do in one useeffect?

File: src/components/home/home.jsx [hunk 3]
```diff
@@ -1,45 +1,83 @@
...
+    const fetchBlogPosts = async () => {
+      try {
+        const response = await fetch('https://dummyjson.com/posts');
+        if (!response.ok) {
+          throw new Error('Failed to fetch blog posts.');
+        }
+        const data = await response.json();
+        console.log("aa", data)
+        setBlogPosts(data.posts.slice(0, 6));
+        setLoading(false);
```
Comment (Mansibansal9648) [inline src/components/home/home.jsx:31]: what is the use of setLoading here?

File: src/components/home/home.jsx [hunk 4]
```diff
@@ -1,45 +1,83 @@
...
 import useCustomDispatch from '../../hooks/useCustomDispatch';
+import { useLocation, useNavigate, Link } from 'react-router-dom';
+import './homePage.css'
 
 function Home() {
-
-  const [user, setUser] = useState({})
+  const [blogPosts, setBlogPosts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
```
Comment (Mansibansal9648) [inline src/components/home/home.jsx:10]: do we need a error state for error or can we directly handle in api calling?

**Comment by amit-negi-23** [comment]
setLoading is for loader that will show at start of website , until the
data is fetched

**Comment by amit-negi-23** [comment]
Yes we can use single useEffect hook but we should prefer separate
useeffect to perform different side effect task as recommended by react

**Comment by amit-negi-23** [comment]
Yes we need error state to rerender the page

**Comment by amit-negi-23** [comment]
Yes we can also define it separately

**Pull request merged**